Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs

```diff
@@ -491,20 +491,9 @@ private async Task RefreshDatabaseAccountInternalAsync(bool forceRefresh)
 
             try
             {
-#nullable disable // Needed because AsyncCache does not have nullable enabled
-                AccountProperties accountProperties = await this.databaseAccountCache.GetAsync(
-                    key: string.Empty,
-                    obsoleteValue: null,
-                    singleValueInitFunc: () => GlobalEndpointManager.GetDatabaseAccountFromAnyLocationsAsync(
-                        this.defaultEndpoint,
-                        this.connectionPolicy.PreferredLocations,
-                        this.GetDatabaseAccountAsync),
-                    cancellationToken: this.cancellationTokenSource.Token,
-                    forceRefresh: true);
-
                 this.LastBackgroundRefreshUtc = DateTime.UtcNow;
-                this.locationCache.OnDatabaseAccountRead(accountProperties);
-#nullable enable
+                this.locationCache.OnDatabaseAccountRead(await this.GetDatabaseAccountAsync(forceRefresh));
```

Comment:
This should always do a force refresh. This is a background task getting the latest account infromation.
```suggestion
                this.locationCache.OnDatabaseAccountRead(await this.GetDatabaseAccountAsync(forceRefresh: true));
```